api/tasks.go

```diff
@@ -931,6 +931,7 @@ type Vault struct {
 	Policies     []string `hcl:"policies,optional"`
 	Role         string   `hcl:"role,optional"`
 	Namespace    *string  `mapstructure:"namespace" hcl:"namespace,optional"`
+	Cluster      *string  `hcl:"cluster,optional"`
```

Comment:
I suppose if we don't really care if the value's set or not (since we `Canonicalize()`) then it makes sense for it to be a `string`, but then it would also be nice if... all the fields received similar treatment?

Reply:
Totally agreed, but unfortunately changing the existing fields would break backwards compatibility for third-party clients. This will need to wait for a hypothetical v2 of the API.